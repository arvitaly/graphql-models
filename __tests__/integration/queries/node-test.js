"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("./../utils");
it("query node", () => __awaiter(this, void 0, void 0, function* () {
    const tester = new utils_1.default();
    const animal = tester.generateModel("animal");
    tester.adapter.findOne.mockReturnValueOnce(animal);
    const result = yield tester.query(`query Q1{
            node(id:"${tester.toGlobalId("Animal", animal.id)}"){
                ... on Animal{
                    name
                }
            }
        }`);
    expect(result).toMatchSnapshot();
}));
