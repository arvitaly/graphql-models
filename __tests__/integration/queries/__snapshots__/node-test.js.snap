// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query node 1`] = `
Object {
  "node": Object {
    "name": "Str1",
  },
}
`;
